spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/support/AbstractRibbonCommand.java

```diff
@@ -93,16 +93,31 @@ protected AbstractRibbonCommand(Setter setter, LBC client,
 				0).get();
 		int commandHystrixTimeout = dynamicPropertyFactory.getIntProperty("hystrix.command." + commandKey + ".execution.isolation.thread.timeoutInMilliseconds",
 				0).get();
-		int ribbonReadTimeout = config == null ? RibbonClientConfiguration.DEFAULT_READ_TIMEOUT :
-				config.get(IClientConfigKey.Keys.ReadTimeout, RibbonClientConfiguration.DEFAULT_READ_TIMEOUT).intValue();
-		int ribbonConnectTimeout = config == null ? RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT :
-				config.get(IClientConfigKey.Keys.ConnectTimeout, RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT).intValue();
-		int ribbonTimeout = ribbonConnectTimeout + ribbonReadTimeout;
+		int ribbonTimeout;
+		if (config == null) {
+			ribbonTimeout = RibbonClientConfiguration.DEFAULT_READ_TIMEOUT + RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT;
+		} else {
+			int ribbonReadTimeout = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".ReadTimeout",
+				config.get(IClientConfigKey.Keys.ReadTimeout, RibbonClientConfiguration.DEFAULT_READ_TIMEOUT)).get();
+			int ribbonConnectTimeout = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".ConnectTimeout",
+				config.get(IClientConfigKey.Keys.ConnectTimeout, RibbonClientConfiguration.DEFAULT_CONNECT_TIMEOUT)).get();
+			//Because ribbon.maxAutoRetries defaults to 0, here we set default Value to 0
+			int maxAutoRetries = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".MaxAutoRetries",
+				config.get(IClientConfigKey.Keys.MaxAutoRetries, 0)).get();
+			//Because ribbon.maxAutoRetriesNextServer defaults to 1, here we set default Value to 1
+			int maxAutoRetriesNextServer = dynamicPropertyFactory.getIntProperty(
+				commandKey + "." + config.getNameSpace() + ".MaxAutoRetriesNextServer",
+				config.get(IClientConfigKey.Keys.MaxAutoRetriesNextServer, 1)).get();
```

Comment:
Instead of `1` you can use `DefaultClientConfigImpl.DEFAULT_MAX_AUTO_RETRIES_NEXT_SERVER`